app/publish/publish.go

```diff
@@ -9,153 +9,95 @@ import (
 	UploadPath string
 }
 
-type UploadOpts struct {
-	Path         string
-	Publisher    Publisher
-	ProxyService *proxy.ProxyService
-}
-
-// NewUploadHandler returns a HTTP upload handler object.
-func NewUploadHandler(opts UploadOpts) (*UploadHandler, error) {
-	var (
-		publisher  Publisher
-		uploadPath string
-	)
-	if opts.ProxyService != nil {
-		publisher = &LbrynetPublisher{ProxyService: opts.ProxyService}
-	} else if opts.Publisher != nil {
-		publisher = opts.Publisher
-	} else {
-		return nil, errors.New("need either a ProxyService or a Publisher instance")
-	}
-
-	if opts.Path == "" {
-		uploadPath = config.GetPublishSourceDir()
-	} else {
-		uploadPath = opts.Path
-	}
-	return &UploadHandler{
-		Publisher:  publisher,
-		UploadPath: uploadPath,
-	}, nil
-}
-
-// Publish takes a file path, account ID and client JSON-RPC query,
-// patches the query and sends it to the SDK for processing.
-// Resulting response is then returned back as a slice of bytes.
-func (p *LbrynetPublisher) Publish(filePath, walletID string, rawQuery []byte) []byte {
-	c := p.ProxyService.NewCaller(walletID)
-	c.SetPreprocessor(func(q *proxy.Query) {
-		params := q.ParamsAsMap()
-		params[fileNameParam] = filePath
-		q.Request.Params = params
-	})
-	r := c.Call(rawQuery)
-	return r
-}
-
 // Handle is where HTTP upload is handled and passed on to Publisher.
 // It should be wrapped with users.Authenticator.Wrap before it can be used
 // in a mux.Router.
-func (h UploadHandler) Handle(w http.ResponseWriter, r *users.AuthenticatedRequest) {
+func (h Handler) Handle(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusOK)
-	if !r.IsAuthenticated() {
-		var authErr Error
-		if r.AuthFailed() {
-			authErr = NewAuthError(r.AuthError)
-		} else {
-			authErr = ErrUnauthorized
-		}
-		w.Write(authErr.AsBytes())
+
+	authResult := auth.FromRequest(r)
+
+	if !authResult.AuthAttempted() {
+		w.Write(proxy.NewAuthRequiredError(errors.New(responses.AuthRequiredErrorMessage)).JSON())
+		return
+	}
+	if !authResult.Authenticated() {
+		w.Write(proxy.NewForbiddenError(authResult.Err()).JSON())
+		return
+	}
+	if authResult.SDKAddress == "" {
+		w.Write(proxy.NewInternalError(errors.New("user does not have sdk address assigned")).JSON())
+		logger.Log().Errorf("user %d does not have sdk address assigned", authResult.User().ID)
 		return
 	}
 
-	f, err := h.saveFile(r)
+	f, err := h.saveFile(r, authResult.User().ID)
 	if err != nil {
 		logger.Log().Error(err)
 		monitor.CaptureException(err)
-		w.Write(NewInternalError(err).AsBytes())
+		w.Write(proxy.NewInternalError(err).JSON())
 		return
 	}
+	defer func() {
+		if err := os.Remove(f.Name()); err != nil {
+			monitor.CaptureException(err, map[string]string{"file_path": f.Name()})
+		}
+	}()
 
-	response := h.Publisher.Publish(f.Name(), r.WalletID, []byte(r.FormValue(JSONRPCFieldName)))
+	w.Write(publish(authResult.SDKAddress, f.Name(), authResult.User().ID, []byte(r.FormValue(jsonRPCFieldName))))
```

Comment:
Many parentheses and calls to very different packages are happening here, I'd unfold it